api/v1alpha2/conversion.go

```diff
@@ -247,15 +254,58 @@ func (dst *MachineDeploymentList) ConvertFrom(srcRaw conversion.Hub) error {
 }
 
 func Convert_v1alpha2_MachineSpec_To_v1alpha3_MachineSpec(in *MachineSpec, out *v1alpha3.MachineSpec, s apiconversion.Scope) error {
+	inGV, err := schema.ParseGroupVersion(in.Bootstrap.ConfigRef.APIVersion)
+	if err != nil {
+		return err
+	}
+
 	if err := autoConvert_v1alpha2_MachineSpec_To_v1alpha3_MachineSpec(in, out, s); err != nil {
 		return err
 	}
 
+	if inGV.Group == bootstrapv1a3.GroupVersion.Group && (out.Bootstrap.ConfigRef.Kind == kubeadmConfigKind || out.Bootstrap.ConfigRef.Kind == kubeadmConfigTemplateKind) {
+		out.Bootstrap.ConfigRef.APIVersion = bootstrapv1a3.GroupVersion.String()
+	}
+
 	// Discards unused ObjectMeta
 
 	return nil
 }
 
+func Convert_v1alpha2_MachineSetSpec_To_v1alpha3_MachineSetSpec(in *MachineSetSpec, out *v1alpha3.MachineSetSpec, s apiconversion.Scope) error {
+	inGV, err := schema.ParseGroupVersion(in.Template.Spec.Bootstrap.ConfigRef.APIVersion)
+	if err != nil {
+		return err
+	}
+
+	if err = autoConvert_v1alpha2_MachineSetSpec_To_v1alpha3_MachineSetSpec(in, out, s); err != nil {
+		return err
+	}
+
+	if inGV.Group == bootstrapv1a3.GroupVersion.Group && (out.Template.Spec.Bootstrap.ConfigRef.Kind == kubeadmConfigKind || out.Template.Spec.Bootstrap.ConfigRef.Kind == kubeadmConfigTemplateKind) {
+		out.Template.Spec.Bootstrap.ConfigRef.APIVersion = bootstrapv1a3.GroupVersion.String()
+	}
+
+	return nil
+}
+
+func Convert_v1alpha2_MachineDeploymentSpec_To_v1alpha3_MachineDeploymentSpec(in *MachineDeploymentSpec, out *v1alpha3.MachineDeploymentSpec, s apiconversion.Scope) error {
+	inGV, err := schema.ParseGroupVersion(in.Template.Spec.Bootstrap.ConfigRef.APIVersion)
+	if err != nil {
+		return err
+	}
+
+	if err = autoConvert_v1alpha2_MachineDeploymentSpec_To_v1alpha3_MachineDeploymentSpec(in, out, s); err != nil {
```

Comment:
```suggestion
	if err := autoConvert_v1alpha2_MachineDeploymentSpec_To_v1alpha3_MachineDeploymentSpec(in, out, s); err != nil {
```

This should be the first call